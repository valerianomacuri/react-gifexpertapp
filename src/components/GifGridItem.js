import React from 'react'

export const GrifGridItem = ({id, title, url}) => {

    return (
        <div className="card animate_animated animate__fadeIn">
            <img src={ url } alt={ title } />
            <p>{ title.toUpperCase() }</p>
        </div>
    )
}