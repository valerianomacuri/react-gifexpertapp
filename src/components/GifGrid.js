import React from 'react'
import PropTypes from 'prop-types'

import { GrifGridItem } from './GifGridItem'
import { useFetchGifs } from '../hooks/useFetchGifs'


export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category )
    
    return (
        <>
            <h3 className="animate_animated animate__fadeIn">{ category }</h3>

            {loading && <p className="animate_animated animate__flash">Loading</p> }

            <div className="card-grid">
                { images.map( img => (
                    <GrifGridItem
                    key={img.id} 
                    {...img } />
                ) )}
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}