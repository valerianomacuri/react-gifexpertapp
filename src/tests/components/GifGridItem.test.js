import { shallow } from "enzyme"
import { GrifGridItem } from "../../components/GifGridItem"

describe('test on GifGridItem', () => {

    const title ='Un titulo'
    const url = 'https://localhost/algo.jpg'
    const wrapper = shallow(<GrifGridItem title={title} url={url} />)

    test('should show component', () => {
        expect(wrapper).toMatchSnapshot()
    })   

    test('should have a paragraph with title', () => {
        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe( title )
    })

    test('should have a url image and alt atributte', () => {
        const img = wrapper.find('img')

        // console.log( img.props().src )
        expect( img.props().src ).toBe( url )
        expect( img.props().alt ).toBe(  title )
    });
    
    test('should have animate__fadeIn class', () => {
        const div = wrapper.find('div')
        
        const className = div.prop('className')
        expect( className.includes('animate__fadeIn') ).toBe( true )

    })
})
