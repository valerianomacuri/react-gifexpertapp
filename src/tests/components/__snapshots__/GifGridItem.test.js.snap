// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifGridItem should show component 1`] = `
<div
  className="card animate_animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
